dart/lib/src/client.dart

```diff
@@ -22,36 +21,14 @@ abstract class SentryClient {
 
   SentryClient.base(
     this.options, {
-    String platform,
-    this.origin,
-    Sdk sdk,
-  })  : _dsn = Dsn.parse(options.dsn),
-        _platform = platform ?? sdkPlatform,
-        sdk = sdk ?? Sdk(name: sdkName, version: sdkVersion);
-
-  final Dsn _dsn;
+    @required this.transport,
+  });
 
   @protected
   SentryOptions options;
 
-  /// The DSN URI.
-  @visibleForTesting
-  Uri get dsnUri => _dsn.uri;
-
-  /// The Sentry.io public key for the project.
-  @visibleForTesting
-  // ignore: invalid_use_of_visible_for_testing_member
-  String get publicKey => _dsn.publicKey;
-
-  /// The Sentry.io secret key for the project.
   @visibleForTesting
-  // ignore: invalid_use_of_visible_for_testing_member
-  String get secretKey => _dsn.secretKey;
-
-  /// The ID issued by Sentry.io to your project.
-  ///
-  /// Attached to the event payload.
-  String get projectId => _dsn.projectId;
+  final Transport transport;
```

Comment:
ideally the transport is read from the `options. transport ` so we don't need to `visibleForTesting`, and we only instantiate a new `transport` here if the current `optins.transport` is a NoOp

Reply:
we can remove this, even for testing we can set a mock/fake on the options

Reply:
✅